deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -608,37 +672,159 @@ def set_module_attribute(self, module_name, attribute, value):
+            self.current_batch_events_size = self.batch_size
+
+        if self.list_events[self.current_batch_events] not in ['network', 'port', 'process'] \
+                or self.current_batch_events_size > 1:
+            event = self.list_events[self.current_batch_events]
         else:
-            self.inventory_path = f"inventory/{self.os}/{self.inventory_sample}"
-        with open(self.inventory_path) as fp:
-            line = fp.readline()
-            while line:
-                if not line.startswith("#"):
-                    msg = "{0}:{1}:{2}".format(self.SYSCOLLECTOR_MQ, self.SYSCOLLECTOR, line.strip("\n"))
-                    self.inventory.append(msg)
-                line = fp.readline()
+            event = self.list_events[self.current_batch_events] + '_end'
+
+        self.current_batch_events_size = self.current_batch_events_size - 1
+        return self.format_event(event)
+
+
+class SCA:
+    def __init__(self, os):
+        self.last_scan_id = 0
+        self.os = os
+        self.count = 0
+        self.SCA_MQ = 'p'
+        self.SCA = 'sca'
+        self.started_time = int(time())
+
+    def get_message(self):
+        if self.count % 100 == 0:
+            msg = self.create_sca_event('summary')
+        else:
+            msg = self.create_sca_event('check')
+        self.count += 1
+
+        msg = msg.strip('\n')
+
+        return f"{self.SCA_MQ}:{self.SCA}:{msg}"
+
+    def create_sca_event(self, event_type):
+        event_data = json.loads('{}')
+        event_data['type'] = event_type
+        event_data['scan_id'] = self.last_scan_id
+        self.last_scan_id += 1
+
+        def create_summary_sca_event(event_data):
+            event_data['name'] = f"CIS Benchmark for {self.os}"
+            event_data['policy_id'] = f"cis_{self.os}_linux"
+            event_data['file'] = f"cis_{self.os}_linux.yml"
+            event_data['description'] = 'This provides prescriptive guidance for establishing a secure configuration.'
+            event_data['references'] = 'https://www.cisecurity.org/cis-benchmarks'
+            total_checks = randint(0, 900)
+            passed_checks = randint(0, total_checks)
+            failed_checks = randint(0, total_checks - passed_checks)
+            invalid_checks = total_checks - failed_checks - passed_checks
+            event_data['passed'] = passed_checks
+            event_data['failed'] = failed_checks
+            event_data['invalid'] = invalid_checks
+            event_data['total_checks'] = total_checks
+            event_data['score'] = 20
+            event_data['start_time'] = self.started_time
+            self.started_time = int(time() + 1)
+            event_data['end_time'] = self.started_time
+            event_data['hash'] = getrandbits(256)
+            event_data['hash_file'] = getrandbits(256)
+            event_data['force_alert'] = '1'
+            return event_data
```

Comment:
```suggestion

            return event_data
```

Reply:
This is not resolved yet.

Reply:
It was solved 5 days ago: https://github.com/wazuh/wazuh-qa/commit/6976fb26ae1e4fcd0d905a309eb1820db1111a5c by @Rebits